### PR TITLE
LPS-121882 Migrate v2 usages in knowledge-base/knowledge-base-web

### PR DESCRIPTION
**Pull request by julien**
In this change we've updated the <clay:management-toolbar-v2 />
to <clay:management-toolbar /> (which uses clay v3).

Creating new knowledge base articles, templates and deleting works as
expected, suggestions can be deleted too.

**Test Plan**:
- Navigate to **Content & Data** > **Knowledge Base**
- Create one or several articles, and delete them
- Create one or several templates and delete them

cc @markocikos, @jonmak08, @ambrinchaudhary

### CODE REVIEW

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/ManagementToolbarPropsTransformer.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+						Liferay.Language.get(
+							'are-you-sure-you-want-to-delete-the-selected-entries'
+						)
+					)
+				) {
+					const form = document.getElementById(
+						`${portletNamespace}fm`
+					);
+
+					if (!form) {
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/ManagementToolbarPropsTransformer.js:33]: See previous comment in https://github.com/liferay-lima/liferay-portal/pull/1507#discussion_r577635010



> See https://github.com/liferay-frontend/liferay-portal/pull/759#discussion_r574501571
> 
> If it's just one line in if block, _submitForm_ in this case, it's hard to justify an early return.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/SuggestionsManagementToolbarPropsTransformer.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+						Liferay.Language.get(
+							'are-you-sure-you-want-to-delete-this'
+						)
+					)
+				) {
+					const form = document.getElementById(
+						`${portletNamespace}fm`
+					);
+
+					if (!form) {
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/SuggestionsManagementToolbarPropsTransformer.js:33]: Same here, see https://github.com/liferay-frontend/liferay-portal/pull/759#discussion_r574501571

**Review by ambrinchaudhary — COMMENTED** [review]
Minor changes for the `if` conditions ;-)

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/ManagementToolbarPropsTransformer.js (hunk 1 above)
Comment (julien) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/ManagementToolbarPropsTransformer.js:33]: I'll do the change, but I think that we have defined so many rules that it is becoming more difficult to remember when to apply them

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/SuggestionsManagementToolbarPropsTransformer.js (hunk 2 above)
Comment (julien) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/SuggestionsManagementToolbarPropsTransformer.js:33]: See https://github.com/liferay-lima/liferay-portal/commit/004fb5e57ce2115242a226090a6d564c4c163a47

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
LGTM. Sorting button is not working but it wasn't working before either, I guess some config is missing. Thx @julien !

**Comment by julien** [comment]
Thanks @carloslancha

**Comment by carloslancha** [comment]
Created https://issues.liferay.com/browse/LPS-128095 to address that bug (`sortingOrder` and/or `sortingURL` are probably not properly configured)

cc @boton @4lejandrito 

Forwarding this one ;)

**Comment by carloslancha** [comment]
ci:forward

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/TemplatesManagementToolbarPropsTransformer.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+					if (!form) {
+						return;
+					}
+
+					form.setAttribute('method', 'post');
+
+					form.querySelector(
+						`#${portletNamespace}kbTemplateIds`
+					).setAttribute(
```
Comment (jonmak08) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/TemplatesManagementToolbarPropsTransformer.js:45]: Do we need to add a `if` check for this DOM node lookup as well?

**Pull request closed without merging**